source/agora/test/EnrollmentManager.d

```diff
@@ -81,7 +81,11 @@ unittest
 
     // check if nodes contains the pre-image previously sent.
     nodes.each!(node =>
-        retryFor(node.hasPreimage(enroll.utxo_key, 999) == true, 5.seconds));
+        retryFor(node.getPreimage(enroll.utxo_key, 999) != PreimageInfo.init,
```

Comment:
Can we compare to the real value here ? If the node returns garbage this test will still pass when it shouldn't.

Reply:
I've already thought about it. But in order to compare to the real value, it must know about the random seed or another preimage. So it needs another unecessary API like getNodePreimage for test.

Reply:
But, I can add checking code for the preimage having garbage.

Reply:
That makes sense. Let's leave it as is for the time being then.